### PR TITLE
handle serviceentry delete

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/19961

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn would it be possible to test this in your env with your test case? If it works, I will add tests.

**Comment by ramaraochavali** [comment]
@howardjohn added tests as well. PTAL

File: pilot/pkg/serviceregistry/external/servicediscovery.go [hunk 1]
```diff
@@ -76,6 +76,13 @@ func NewServiceDiscovery(configController model.ConfigStoreCache, store model.Is
 				}
 			}
 
+			// If service entry is deleted, cleanup endpoint shards for services.
+			if event == model.EventDelete {
+				for _, svc := range cs {
+					c.XdsUpdater.SvcUpdate(c.Cluster(), string(svc.Hostname), svc.Attributes.Namespace, event)
+				}
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/external/servicediscovery.go]: return on delete?

Comment (ramaraochavali) [reply pilot/pkg/serviceregistry/external/servicediscovery.go]: It should trigger full push to delete these clusters and push to Envoys. `SvcUpdate` just deletes the shards. right?

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/external/servicediscovery.go]: ah, i missed it

**Comment by hzxuzhonghu** [comment]
I have one thought in mind: is it feasible we make both ServceEntry and kube Service/Endpoints handler consistent to prevent diverge?

**Comment by ramaraochavali** [comment]
> I have one thought in mind: is it feasible we make both ServceEntry and kube Service/Endpoints handler consistent to prevent diverge?

I agree this would be useful.

**Pull request merged**